Linux/sgx/libsgx_tsgxssl/tpthread.cpp

```diff
@@ -63,9 +57,9 @@ int sgxssl_pthread_once (pthread_once_t *once_control, void (*init_routine) (voi
 
 	sgx_spin_lock(&pthread_once_lock);
 
-	if (*once_control != ONCE_CONTROL_INIT &&
-		*once_control != ONCE_CONTROL_COMPLETE &&
-		*once_control != ONCE_CONTROL_BUSY) {
+	if (once_control->state != ONCE_CONTROL_INIT &&
```

Comment:
The once_control actually is an integer from OpenSSL because OpenSSL is using system definition. Suggest to undef  current pthread_once_t from SGX pthread.h and then redefine it as an integer.